### PR TITLE
Port greggman new program handling test

### PR DESCRIPTION
**Pull request by shrekshao**
Mostly a port from @greggman 's program handling test https://github.com/KhronosGroup/WebGL/issues/2842

There is a section doing similar tests in `program-test.html`. But since it only detects gl errors without actually drawing something, it couldn't really expose problems. I removed that one test case from there. The new `program-handling.html` tests covers better for the topic of program handling for various linking situation.

### CODE REVIEW

File: sdk/tests/conformance/programs/program-handling.html [hunk 1]
```diff
@@ -0,0 +1,143 @@
+<!--
+
+/*
+** Copyright (c) 2019 The Khronos Group Inc.
```
Comment (jdashg) [inline sdk/tests/conformance/programs/program-handling.html]: There's a new shorter copyright block:
https://github.com/KhronosGroup/WebGL/pull/2846/files

File: sdk/tests/conformance/programs/program-test.html [hunk 2]
```diff
@@ -331,10 +331,6 @@
     assertMsg(gl.getProgramParameter(progGood2, gl.LINK_STATUS) == false,
               "linking should fail with in-use formerly good program, with new bad shader attached");
 
-    // Invalid link leaves previous valid program intact.
```
Comment (jdashg) [inline sdk/tests/conformance/programs/program-test.html]: I suppose we can remove this in favor of the more detailed tests, but I generally don't like removing tests like this.

Comment (shrekshao) [reply sdk/tests/conformance/programs/program-test.html]: So you prefer to leave the old test unchanged (restore the deleted section)? (Just to double check I understand correctly)

Comment (jdashg) [reply sdk/tests/conformance/programs/program-test.html]: I prefer leaving the old test, but I don't mind strongly.

File: sdk/tests/conformance/programs/00_test_list.txt [hunk 3]
```diff
@@ -6,6 +6,7 @@ gl-get-active-uniform.html
 gl-getshadersource.html
 gl-shader-test.html
 invalid-UTF-16.html
+program-handling.html
```
Comment (jdashg) [inline sdk/tests/conformance/programs/00_test_list.txt]: `--min-version 1.0.4` for new tests, sorry!

File: sdk/tests/conformance/programs/program-test.html (hunk 2 above)
Comment (shrekshao) [reply sdk/tests/conformance/programs/program-test.html]: I have one more reason to delete this as current chromium implementation didn't handle this properly. Separate this part of testing into the new program-handling.html would make suppression a bit easier for us. So I think I will keep it removed.

File: sdk/tests/conformance/programs/00_test_list.txt (hunk 3 above)
Comment (shrekshao) [reply sdk/tests/conformance/programs/00_test_list.txt]: Thanks for the instruction

File: sdk/tests/conformance/programs/program-test.html (hunk 2 above)
Comment (jdashg) [reply sdk/tests/conformance/programs/program-test.html]: I don't like this sort of gamesmanship. Please focus on fixing the failing subtests, and don't focus on overall pass rate.

Comment (shrekshao) [reply sdk/tests/conformance/programs/program-test.html]: FYI I'm working on fixing the handling issue at the same time.

Comment (shrekshao) [reply sdk/tests/conformance/programs/program-test.html]: Restored.

**Comment by jdashg** [comment]
Thanks!

**Comment by kenrussell** [comment]
Nice work Shrek. LGTM after the fact.

**Comment by kainino0x** [comment]
Sorry if I'm misunderstanding, I'm not following this closely. If I understood correctly, the point of having the failure in a separate test was not to improve pass rate, but to allow us to suppress the failure without losing test coverage of the whole file. Were things landed in a way that doesn't lose us test coverage?

**Comment by shrekshao** [comment]
Thanks Kai for clarifying this. Jeff would this change your mind? I could
open a separate pull request

**Comment by jdashg** [comment]
As you can see in this PR, it only lands the new test. Pre-existing tests are not modified, so suppression hasn't changed here.
Google is free to modify the conformance test checkout they use for CI. (Firefox has definitely done this in the past)
That, or fixing this incorrect behavior, should be the primary approaches.

**Comment by kainino0x** [comment]
Ok, it doesn't affect the existing coverage. We don't normally modify our conformance test checkout - our workflow would make this hard and it would make pulling CTS updates more complex.

I'm just talking about taking exactly the existing tests and splitting them into two files so the suppressions can be more targeted. If I understood correctly, in this case that would improve our coverage.

Anyway, I'm not dead set on it, but I think that being able to have more targeted suppressions is always better.

**Pull request merged**